components/brave_wallet/common/config.gni

```diff
@@ -10,4 +10,5 @@ declare_args() {
   brave_infura_project_id = ""
   enable_bitcoin_by_default = !is_ios
   enable_zcash_by_default = !is_ios && !is_android
+  enable_orchard = brave_channel == "nightly" || brave_channel == "development"
```

Comment:
I think actually using this buildflag here is problematic because it makes it problematic to add tests for this new functionality. The existing tests will only run on channel builds and that is not sufficient. I think this should just be a feature flag and enable per channel through griffin.

Reply:
I think this is not a good idea because we will get orchard crate into builds and increase binary size without giving any functions

Reply:
Do we run tests on non-debug builds?

Reply:
Another thing is i believe we don't want to give an access to unfinished feature related to money even it is under the feature flag since this may lead to user money lose.

Reply:
Huh? We run tests on every single PR, but these will not run on them so someone can break things and not find out until it hits nightly channel. Also there is no access to it when the feature is disabled. I'm not suggesting it should be in `about://flags`. This is not acceptable to have tests that only run for release channel builds.

Reply:
Even locally people will have to do an official build and have to know they need to do an official build for these tests to run.

Reply:
Technically we could modify the conditions to also include non official builds, but then we have the possibly even worse situation where we don't know if the buildflag guards are used correctly until it hits release channel and maybe fails there.

Reply:
> Literally every feature that is not always enabled by default adds to the binary size without providing any functionality. Enabling something per-channel is not an appropriate use of buildflags

This definitely needs a sign off from the security team to enable such a big amount of third-party code in the stable\beta channel by default. 
cc @kylehickinson

Reply:
as the feature potentially introduces noticeable increase of binary size (esp. on android / ios) we need to somehow keep it under control and explicit for the android and ios teams (as each change of the binary size affects business metrics, such as downloads) and also perf team as it will trigger the APK size perf test.

We could keep the enabled buildflag (I also agree that it shouldn't be conditionally on for certain channels) so that we will be able to turn it off and measure the apk size difference

Reply:
Let's make it a default true, to allow measuring of actual size change in the future

Reply:
> This definitely needs a sign off from the security team to enable such a big amount of third-party code in the stable\beta channel by default.

I'm good to sign off on this. It's pretty normal for us to do it that way and since we've already reviewed the 3p dependencies and plan to use them there's no concerns from me if we just want to gate this on a feature flag only.

Reply:
Regarding app size: i've build 3 arm64 apks:
[719f2cf7d6a6d6ec62395c2ae3d7445290947899](https://ci.brave.com/job/test-brave-browser-build-android-variant/1310/) - baseline - 386,2 mb
[e1f9c2b2579d19b729adf13ea88c89a5fcc5e773](https://ci.brave.com/job/test-brave-browser-build-android-variant/1311/) - this pr, orchard + keyring - 386,3 mb
[237ca6fd9905d08e945f304c71ec38f65380f4c1](https://ci.brave.com/job/test-brave-browser-build-android-variant/1312/) - orchard + keyring + shielding - 387,5 mb
Seems it doesn't bring too much size for now, but i'd still keep the flag for future measures